src/Period/PeriodAbstract.php

```diff
@@ -117,12 +117,7 @@ public function includes(PeriodInterface $period, $strict = true)
      */
     public function containsEvent(EventInterface $event)
     {
-        return
-            $event->containsPeriod($this) ||
-            $event->isDuring($this) ||
-            $this->contains($event->getBegin()) ||
-            ($event->getEnd() && $this->contains($event->getEnd())  && $event->getEnd()->format('c') !== $this->begin->format('c'))
-        ;
+        return $event->getBegin() < $this->getEnd() && $event->getEnd() > $this->getBegin();
```

Comment:
Do you have any unit test about this expected behavior is same as actual?

I think it needs some tests to present that this can fix the timezone issue :).